bc_obps/registration/tests/models/test_contact.py

```diff
@@ -36,3 +37,70 @@ def setUpTestData(cls):
+        )
+        self.senior_officer_contact = baker.make_recipe(
+            "registration.tests.utils.contact", business_role=BusinessRole.objects.get(role_name='Senior Officer')
+        )
+        self.null_address_contact = baker.make_recipe("registration.tests.utils.contact", address=None)
+
+    def test_prevent_address_null_for_op_rep(self):
+        """Test that updating address to NULL fails for Operation Representative."""
+        with transaction.atomic():
+            with self.assertRaises(ProgrammingError) as cm:
+                self.op_rep_contact.address = None
+                self.op_rep_contact.save()
+
+            self.assertIn(
+                "Cannot set address to NULL for a contact with role Operation Representative", str(cm.exception)
+            )
+
+        # Verify the address wasn’t changed
+        self.op_rep_contact.refresh_from_db()
+        self.assertIsNotNone(self.op_rep_contact.address, "Address was set to NULL for Operation Representative")
+
+    def test_allow_address_null_for_non_op_rep(self):
+        """Test that updating address to NULL succeeds for non-Operation Representative."""
+        self.senior_officer_contact.address = None
+        self.senior_officer_contact.save()
+
+        # Verify the address is now NULL
+        self.senior_officer_contact.refresh_from_db()
+        self.assertIsNone(self.senior_officer_contact.address)
+
+    def test_allow_address_change_to_non_null_for_op_rep(self):
+        """Test that updating address to a different non-NULL value succeeds for Operation Representative."""
+        new_address = baker.make_recipe("registration.tests.utils.address")
+        self.op_rep_contact.address = new_address
+        self.op_rep_contact.save()
+
+        # Verify the address was updated
+        self.op_rep_contact.refresh_from_db()
+        self.assertEqual(self.op_rep_contact.address, new_address)
+
+    def test_no_trigger_on_other_field_update(self):
+        """Test that updating a non-address field doesn’t trigger the restriction."""
+        self.op_rep_contact.email = "new_op_rep@example.com"
+        self.op_rep_contact.save()
+
+        # Verify the email was updated and address unchanged
+        self.op_rep_contact.refresh_from_db()
+        self.assertEqual(self.op_rep_contact.email, "new_op_rep@example.com")
+        self.assertIsNotNone(self.op_rep_contact.address)
+
+    def test_no_trigger_when_address_already_null(self):
+        """Test that updating a contact with NULL address doesn’t trigger if address stays NULL."""
+        # Update another field
+        self.null_address_contact.email = "updated_null_op_rep@example.com"
+        self.null_address_contact.save()
+
+        # Verify the update succeeded and address is still NULL
+        self.null_address_contact.refresh_from_db()
+        self.assertEqual(self.null_address_contact.email, "updated_null_op_rep@example.com")
+        self.assertIsNone(self.null_address_contact.address)
```

Comment:
Similar to above question, do we need to allow this case? If someone somehow got away with giving us a null address, we'd probably want to them fix it

Reply:
I have changed that part of the trigger so this test is no longer valid.

Reply:
Test reverted 😃 
https://github.com/bcgov/cas-registration/pull/3052#discussion_r2019423057